docs/modules/ROOT/pages/crosschain.adoc

```diff
@@ -0,0 +1,175 @@
+= Adding cross-chain support to contracts
+
+If your contract is targetting to be used in the context of multichain operations, you may need specific tools to identify and process these cross-chain operations.
+
+OpenZeppelin provides the xref:api:crosschain.adoc#CrossChainEnabled[`CrossChainEnabled`] abstract contract, that includes dedicated internal functions.
+
+In this guide, we will go through an example use case: _how to build an upgradeable & mintable ERC20 token controlled by a governor present on a foreign chain_.
+
+== Starting point, our ERC20 contract
+
+Let's start with a small ERC20 contract, that we bootstrapped using the https://wizard.openzeppelin.com/[OpenZeppelin Contracts Wizard].
+
+[source,solidity]
+----
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+
+contract MyToken is Initializable, ERC20Upgradeable, UUPSUpgradeable {
+    address public owner;
+
+    modifier onlyOwner() {
+        require(owner == _msgSender(), "Not authorized");
+        _;
+    }
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() initializer {}
+
+    function initialize(address initialOwner) initializer public {
+        __ERC20_init("MyToken", "MTK");
+        __UUPSUpgradeable_init();
+
+        owner = initialOwner;
+    }
+
+    function mint(address to, uint256 amount) public onlyOwner {
+        _mint(to, amount);
+    }
+
+    function _authorizeUpgrade(address newImplementation) internal override onlyOwner {
+    }
+}
+----
+
+This token is mintable and upgradeable by the owner of the contract.
+
+== Preparing our contract for cross-chain operations.
+
+Let's now imagine that this contract is going to live on one chain, but we want the minting and the upgrading to be performed by a xref:governance.adoc[`governor`] contract on another chain.
+
+For example, we could have our token on xDai, with our governor on mainnet, or we could have our token on mainnet, with our governor on optimism
+
+In order to do that, we will start by adding xref:api:crosschain.adoc#CrossChainEnabled[`CrossChainEnabled`] to our contract.
+
+```diff
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
++import "@openzeppelin/contracts-upgradeable/crosschain/CrossChainEnabled.sol";
+
+-contract MyToken is Initializable, ERC20Upgradeable, UUPSUpgradeable {
++abstract contract MyTokenCrossChain is Initializable, ERC20Upgradeable, UUPSUpgradeable, CrossChainEnabled {
+```
+
+Once that is done, we can use the `onlyCrossChainSender` modifier, provided by `CrossChainEnabled` in order to protect the minting and upgrading operations.
+
+```diff
+-    modifier onlyOwner() {
+-        require(owner == _msgSender(), "Not authorized");
+-        _;
+-    }
+
```

Comment:
This isn't core to the point we want to make so I would just remove it to minimize noise.

```suggestion
```